fuzzing/replay/test_file_buffer.h

```diff
@@ -0,0 +1,71 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef FUZZING_REPLAY_TEST_FILE_BUFFER_H_
+#define FUZZING_REPLAY_TEST_FILE_BUFFER_H_
+
+#include <cassert>
+#include <cstddef>
+#include <memory>
+#include <string>
+
+#include "absl/status/status.h"
+#include "absl/strings/string_view.h"
+
+namespace fuzzing {
+
+// A memory buffer for reading and storing test data from files.
+//
+// This class permits the use of a single buffer to store the data from multiple
+// test files read sequentially, resulting in reduced memory churn for large
+// corpora consisting of multiple files.
+class TestFileBuffer {
+ public:
+  // Creates a new buffer holding up to `max_size` bytes.
+  TestFileBuffer(size_t max_size)
+      : max_size_(max_size), last_size_(0), buffer_(new char[max_size]) {
```

Comment:
nit: with negative max_size, new char[max_size] will throw an exception with minimal information. Perhaps we can initialize it later than the assert?

Reply:
Oh that's a great point - fixed!